deploy_nixos/nixos-deploy.sh

```diff
@@ -66,13 +70,28 @@ targetHostCmd() {
 # Ensure the local SSH directory exists
 mkdir -m 0700 -p "$HOME"/.ssh
 
-# Build derivation
-log "building nix code"
-outPath=$(nix-store --realize "$drvPath" "${buildArgs[@]}")
+if [[ "${buildOnTarget:-false}" == true ]]; then
+
+  # Upload derivation
+  log "uploading derivations"
+  copyToTarget "$drvPath" --substitute-on-destination
+
+  # Build remotely
+  log "building on target"
+  set -x
+  targetHostCmd "nix-store" "--realize" "$drvPath" "${buildArgs[@]}"
```

Comment:
It's starting to look a lot like the `nixos-rebuild` script. I wonder if that couldn't be used instead.

Reply:
`nixos-rebuild` works with remote builders rather than drv upload, therefore requiring `trusted-user`. If you don't have that, that's a reason to prefer this method.